dns/zone-configs/k8s.io.yaml

```diff
@@ -32,6 +32,14 @@ canary:
   - ns-cloud-c3.googledomains.com.
   - ns-cloud-c4.googledomains.com.
 
+go.canary:
```

Comment:
Why do we need this.  Why is it not sufficent to delegate `go` which implicitly does `go.canary` before going live?

Reply:
This was suggested to be done to be more explicit. so I followed that request.

Reply:
Happy to remove this.

Reply:
if I understand it properly, this is not necessary.  Do I understand?  When we push to canary.k8s.io, we should be able to test go.canary.k8s.io, distinct from go.k8s.io - right?  We should add tests to the DNS test to ensure that canaries are correct.

Reply:
As far as I understand it with the explicit addition here it would enable a separate zone within google cloud dns.
But happy to drop this.
And yes we should be able to test go.canary.k8s.io distinctly from go.k8s.io. as we are already within a separate zone.